### ISSUE TITLE
Pass tags to completions filter function in org-roam-find-file?

### ISSUE THREAD

**Issue by aldanor** [org-roam/org-roam#1410]
### Brief Abstract

If I understand correctly (apologies in advance if I'm misunderstanding something), `org-roam-find-file` takes a filter function and provides title/path to it, whereas it could also easily provide tags **without any cost** (because they are already extracted).

This would allow e.g., building your own `org-roam-find-file-with-tag-foo` functions in a clean way without having to parse post-processed completions again (or even more generally, something like`org-roam-find-tagged-file`).

### Long Description

`org-roam--get-title-path-completions` currently does this:

https://github.com/org-roam/org-roam/blob/fde40dc1c43fcb34cde3253127c3cc27abe7f1a7/org-roam.el#L865-L867

So, basically, it has tags at hand but it discards them after merging them into completion strings.

### Proposed Implementation (if any)

```elisp
        (let ((k (org-roam--add-tag-string title tags))
              (v (list :path file-path :title title :tags tags)))
          (push (cons k v) completions))))))
```

### Please check the following:

- [x] No similar feature requests

**Comment by d12frosted** [comment]
I totally agree that in some cases more information is needed during filtering phase.

@jethrokuan was it planned as part of #1401 (so these filters accept a node)? If not, do you mind PR for this?

Since I thought that this idea is not popular, I ended up with using [vulpea][1] that allows me to use `tags`, `title`, `id`, `level` and `path` in the filter function:

```emacs-lisp
  (vulpea-select
   "Wine"
   nil nil
   (lambda (note)
     (let ((tags (vulpea-note-tags note)))
       (and (seq-contains-p tags "wine")
            (seq-contains-p tags "cellar")))))
```

  [1]: https://github.com/d12frosted/vulpea

**Comment by jethrokuan** [comment]
My preferred way of implementing these filters is by adding WHERE clauses in the SQL queries. I'd love to support this cleanly, but this issue is a blocker for now:
https://github.com/org-roam/org-roam/pull/1401#issuecomment-768267497

There's no clean way to retrieve tags from the normalized tags table in Emacsql. This means obtaining the completions require fetching all the completions, and then looping over them and filtering and doing a group-by via elisp.

If I release v2 while still using emacsql, then this will be supported via the proposed implementation above.

**Comment by d12frosted** [comment]
@jethrokuan 
Aye, with `where` clauses it would be more efficient (though even with 3.5k notes simple filtering works fast enough).

I will take a closer look at your comments later. As well as UX question :) Thanks for sharing your thoughts 💯

**Comment by jethrokuan** [comment]
In v2, `org-roam-node-read`'s `filter-fn` supports filtering by tags.
